### DISCUSSION TITLE
Merged memory loop is not repeating

### DISCUSSION THREAD

**Discussion by jaapnoordzij** [pschatzmann/arduino-audio-tools#1471] [Q&A] [answered]
Hi Phil,

I am in a learning curve following many of your examples, changing bits and things ... and learning ... In your audiokit example "streams-memory_pcm-mixer-audiokit" you mix a "drums" and a "guitar" memory stream and the stream is in an endless loop. 
I have tried to replace the InputMixer with an InputMerger.

To obtain the effect, the only thing I did is change the number of channels from 1 to 2 in the kit configuration.
It works as intended - I can hear the drums left and the guitar right all perfectly - but the streams are no longer repeating. 
What could be the explanation for this ?

```
InputMerge<int16_t> merger;
AudioBoardStream kit(AudioKitEs8388V1);
MemoryStream drums(drums_raw, drums_raw_len);
MemoryStream guitar(guitar_raw, guitar_raw_len);
StreamCopy copier(kit, merger);
const int sample_rate = 8000;
```
In Setup:
```
  // auto restart when MemoryStream has ended
  drums.setLoop(true);
  guitar.setLoop(true);

  // setup output
  auto cfg = kit.defaultConfig(TX_MODE);
  cfg.channels = 2;
  cfg.sample_rate = sample_rate;
  kit.begin(cfg);
  // max volume
  kit.setVolume(1.0);

  merger.add(drums);
  merger.add(guitar);
  merger.begin(cfg);
```

**Comment by pschatzmann** [comment, marked as answer]
Interesting question: the copier checks for the available data and in the InputMerge I use the min of all streams to provide this result.

I changed the available() method in MemoryStream to return a fixed DEFAULT_BUFFER_SIZE instead of the effective available bytes up to the end when it is in loop mode. With this change, your sketch should work correctly now!

The correction has been committed to github
